src/scripts/main.js

```diff
@@ -1,3 +1,34 @@
 'use strict';
 
-// write code here
+const table = document.querySelector('table');
+const tableHead = [...table.tHead.firstElementChild.cells];
+const tableRows = [...table.tBodies[0].rows];
+
+table.addEventListener('click', (event) => {
+  if (tableHead.includes(event.target)) {
+    const sortString = (a, b) => {
+      return b < a ? 1 : -1;
+    };
+
+    const sortNumber = (a, b) => {
+      return +(a.replace(/[^+\d]/g, '')) - +(b.replace(/[^+\d]/g, ''));
+    };
+
+    const selectedRow = event.target.innerText;
+
+    return (selectedRow === 'Age' || selectedRow === 'Salary')
+      ? sortTable(event.target, sortNumber)
+      : sortTable(event.target, sortString);
+  }
+});
+
+function sortTable(header, compareFunction) {
+  const indexOfColumn = tableHead.indexOf(header);
+
```

Comment:
I like that you create a separate function for sorting, but let's refactor this function a little bit - instead of creating and then passing compare function for each single case (sortString, sortNumber), let move all the checks inside the sortTable function

Reply:
this way, we'll have a single function responsible for table sorting